src/mitol/common/decorators.py

```diff
@@ -27,3 +30,48 @@ def _cache_controlled(request, *args, **kw):
         return _cache_controlled
 
     return _cache_controller
+
+
+def single_task(
+    timeout: int,
+    raise_block: Optional[bool] = True,
+    lock_str: Optional[str] = None,
```

Comment:
I'd mentioned adding an option to pass a callable for more flexibility on the key generation. I'm thinking instead of `lock_str` here we'd follow the convention of python's `sorted` `key=` arg.

Earlier in imports:

```python
from typing import ParamSpec
P = ParamSpec('P')
```

Change:

```suggestion
    key: Optional[str | Callable[str, P.args, P.kwargs]] = None,
```

Then:

```python
if isinstance(key, str):
    lock_id = key
elif callable(key):
    lock_id = key(func.__name__, args, kwargs)
else:
    lock_id = func.__name__
```

I've dropped the `args[0] if args else 'single'` bit here because I feel that `-single` doesn't really provide anything additional here and doing `str()` on an arbitrary first positional arg feels a bit too automagical for my tastes and if the argument's `__str__` isn't unique could cause issues that might not get caught until production.

You many need to add `typing-extensions` as a requirement for python <3.10 to get `ParamSpec` since this project supports those.